### PR TITLE
Revamp oauth scope handling; support requiring multiple scopes.

### PR DESCRIPTION
**Pull request by inklesspen**
This code should properly handle a situation where multiple scopes are necessary for access.

### CODE REVIEW

File: endpoints/users_id_token.py [hunk 1]
```diff
@@ -316,6 +316,25 @@ def _set_oauth_user_vars(token_info, audiences, allowed_client_ids, scopes,
 # pylint: enable=unused-argument
 
 
+def _process_scopes(scopes):
```
Comment (kryzthov) [inline endpoints/users_id_token.py]: Maybe document what `scopes` is?
I believe `scopes` should be a list of string, and each string is a space-separated list of scope.
Is this correct?

File: endpoints/users_id_token.py [hunk 2]
```diff
@@ -327,27 +346,27 @@ def _set_bearer_user_vars(allowed_client_ids, scopes):
...
-      _logger.warning('Client ID is not allowed: %s', client_id)
-      return
+  all_scopes, sufficient_scopes = _process_scopes(scopes)
+  try:
+    authorized_scopes = oauth.get_authorized_scopes(sorted(all_scopes))
+  except oauth.Error:
+    _logger.debug('Unable to get authorized scopes.', exc_info=True)
+    return
+  if not _are_scopes_sufficient(authorized_scopes, sufficient_scopes):
+    _logger.debug('Authorized scopes did not satisfy scope requirements.')
```
Comment (kryzthov) [inline endpoints/users_id_token.py]: Shouldn't we return or fail here?

Also, would it be useful to include the actual authorized/sufficient scopes in the log message?

File: endpoints/users_id_token.py (hunk 1 above)
Comment (inklesspen) [reply endpoints/users_id_token.py]: Oh yeah, whoops!

File: endpoints/users_id_token.py (hunk 2 above)
Comment (kryzthov) [reply endpoints/users_id_token.py]: Fair enough

File: endpoints/users_id_token.py [hunk 3]
```diff
@@ -327,27 +364,28 @@ def _set_bearer_user_vars(allowed_client_ids, scopes):
...
+  if (list(allowed_client_ids) != SKIP_CLIENT_ID_CHECK and
+      client_id not in allowed_client_ids):
+    _logger.warning('Client ID is not allowed: %s', client_id)
     return
 
-  _logger.debug('Oauth framework user didn\'t match oauth token user.')
-  return None
+  os.environ[_ENV_USE_OAUTH_SCOPE] = ' '.join(authorized_scopes)
+  _logger.debug('Returning user from matched oauth_user.')
+  return
```
Comment (kryzthov) [inline endpoints/users_id_token.py]: I guess this return isn't strictly necessary.

**Pull request merged**